### PR TITLE
Site building in Travis

### PR DESCRIPTION
**Pull request by bebatut**
Hi,

I wanted to add several plugins to Jekyll, for example to manage the bibliography or the figure caption (next PR...).
But as the website is currently building by GitHub, we are limited to the plugins approved by GitHub and we can not custom ones.

One way to overpass that is to use Travis to build the website, push it to a `gh-page` branch. and tell to GitHub to serve this branch. 

I then change a bit Travis to do that. I tested it on my fork: http://bebatut.fr/training-material/

And because we are building the site on Travis, we can even use the `make check` command of the (still open) PR #424 in `.travis.yml` to check the correct building of the site and checking the links before pushing it.

Bérénice

### CODE REVIEW

**Comment by shiltemann** [comment]
very neat :) 

I'll merge the htmlproofer PR (I started working on a htmlproofer command to replace vl in travis last week but ran into some upstream bugs (for example by default it complains if images don't have alt-texts, which I'm not sure we want to force people to have to bother with, but their flag to ignore these errors is not working, and some other small things, so maybe we keep vl in travis until we have a good replacement command using htmlproofer? (definitely nice to to this because it can handle relative paths and vl cannot)

**Comment by bebatut** [comment]
@shiltemann It could be great that most of the images have a alt text (for accessibility) and also because we worked on a plugin to generate automatically caption for the figure out of the alt text or title text directly in the markdown (doc here: https://github.com/bebatut/training-material/blob/figure_caption/topics/training/tutorials/create-new-tutorial-content/tutorial.md)
I am just waiting for this PR to be merged to create the next one :)

**Comment by shiltemann** [comment]
ah cool, very nice :)

File: bin/publish [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+git config user.name "Travis-CI"
+git config user.email "noreply@travis-ci.org"
+git remote rm origin
+git remote add origin "https://bebatut:$GH_TOKEN@github.com/bebatut/training-material"
```
Comment (willdurand) [inline bin/publish]: This must be updated to push to the `galaxyproject` I guess.

Comment (willdurand) [inline bin/publish]: Also, note that you could do `set-url` instead of `rm + add`

File: topics/transcriptomics/tutorials/ref-based/tutorial.md [hunk 2]
```diff
@@ -144,7 +144,7 @@ Spliced mappers have been developed to efficiently map transcript-derived reads
 2. Generation of possible splices between neighboring exons
 3. Comparison of reads that did not initially map to the genome against these *in silico* created junctions
 
-![](../../images/tophat2.png)
+![Kim et al.](../../images/tophat2.png "Kim et al., Genome Biology, 2013")
```
Comment (willdurand) [inline topics/transcriptomics/tutorials/ref-based/tutorial.md]: The changes in this file should not be part of this PR.

**Comment by willdurand** [comment]
LGTM

**Comment by shiltemann** [comment]
could we put the vl link checking command back into travis before we merge? just until we can replace it with the htmlproofer check? (right now getting 700+ errors because existing materials mostly have no alt-texts in images and am a bit afraid people will then just ignore this check completely and more broken links could sneak in in the meantime)

other than that, yes! this is great, let's get it in :)

**Comment by bebatut** [comment]
@shiltemann I forgot that... Fixed now

**Comment by shiltemann** [comment]
thanks a lot @bebatut!

**Comment by bebatut** [comment]
Thanks @shiltemann and @willdurand for the help here
